src/olympia/devhub/tasks.py

```diff
@@ -187,9 +187,13 @@ def handle_upload_validation_result(results, upload_pk, channel):
     FileUpload instance."""
     upload = FileUpload.objects.get(pk=upload_pk)
 
-    if not upload.addon_id:
+    if (not upload.addon_id or
+            not upload.addon.find_latest_version(channel=channel, exclude=())):
```

Comment:
Do we want to consider the edge case of an add-on that has all the versions deleted (in a single channel, or both channels)?